packages/router/test/integration.spec.ts

```diff
@@ -76,6 +76,52 @@ describe('Integration', () => {
          ]);
        })));
 
+    it('should allow filtering of routing events', fakeAsync(inject([Router], (router: Router) => {
+         router.onSameUrlNavigation = 'reload';
+         router.resetConfig([
+           {path: '', component: SimpleCmp},
+           {path: 'simple', component: SimpleCmp},
+         ]);
+
+         const fixture = createRoot(router, RootCmp);
+         const events: Event[] = [];
+         router.events.subscribe(e => events.push(e));
+
+         router.navigateByUrl('/simple');
+         tick();
+
+         // Verify that all events are published by default
+         expectEvents(events, [
+           [NavigationStart, '/simple'],
+           [RoutesRecognized, '/simple'],
+           [GuardsCheckStart, '/simple'],
+           [ChildActivationStart],
+           [ActivationStart],
+           [GuardsCheckEnd, '/simple'],
+           [ResolveStart, '/simple'],
+           [ResolveEnd, '/simple'],
+           [ActivationEnd],
+           [ChildActivationEnd],
+           [NavigationEnd, '/simple'],
+         ]);
+
+         events.length = 0;
```

Comment:
```suggestion
         events = [];
```

To be consistent with other tests I've seen in the code base (and a little more obvious to people that don't know the "length zero" trick)